### DISCUSSION TITLE
Tab instead of Enter to use code shortcuts?

### DISCUSSION THREAD

**Discussion by 7ddgaming** [phcode-dev/phoenix#1809] [Q&A] [answered]
Hi everyone, I want to use Tab instead of Enter (default) to complete code shortcuts while typing like in VSCode.  However, I can't find an option in Keyboard Mapping.  I don't want to change Enter's primary function to go to the next line, which seems like the only setting in the Key-mapping setting.  Also checked the JSON settings file and didn't see anything there, either, unless I missed something.  Any advice would be appreciated.  Thank you!

Brandon

**Comment by abose** [comment]
Hi @7ddgaming download the emmet extension from the extension store. If that is not what you want can you comment the exact thing to do when you press tab? Most likely the emmet extension should help.

**Reply by 7ddgaming** [reply]
> Hi @7ddgaming download the emmet extension from the extension store. If that is not what you want can you comment the exact thing to do when you press tab? Most likely the emmet extension should help.

I'd like to expand the abbreviation like if I type "document.getElem" and I want to be able to hit Tab instead of Enter key to finish the syntax automatically.  I downloaded Emmet and I see the "Expand abbreviations with Tab key" is checked, but it still isn't working.  I restarted the app as well.
![Untitled](https://github.com/user-attachments/assets/1ab8d83d-03e6-4cd9-87bc-e43a9a5e4142)

**Reply by abose** [reply]
Ah, you meant selecting using tab from the auto complete hint list instead of enter key?  

If so, let me see if i can add that feature in the next release. Emmet only works for html files to do Emmet things.

**Reply by 7ddgaming** [reply]
> Ah, you meant selecting using tab from the auto complete hint list instead of enter key?
> 
> If so, let me see if i can add that feature in the next release. Emmet only works for html files to do Emmet things.

Yep!  I'm used to using Tab to do that from other software, but love Phoenix code.  There are probably others that feel the same way too.  Especially if they come from VSCode.  I'll use AutoHotKey in the meantime.  Thanks a lot and good luck with your extension!

**Reply by abose** [reply]
Looks like its Good to add as a default behavior in phoenix itself. Thanks for the feature request.

**Reply by abose** [reply, marked as answer]
@7ddgaming Fixed via https://github.com/phcode-dev/phoenix/pull/1816
This should be avilable as the default behaior in the next release (V3.9+). 

If you want to make it working in the current release(releases less than 3.8.9), select menu `File> Open Preferences File`, then on the right at the end of the file `phcode.json`, add this text `"insertHintOnTab": true` like so and save:
![image](https://github.com/user-attachments/assets/06cf2036-952b-4319-bfd9-cfe553de1895)

Close the preferences by selecting `no-split` from the dropdown 
![image](https://github.com/user-attachments/assets/d9ec20d4-f60d-4c58-a3bb-0c0cde5c565d)
